app/src/unitTest/kotlin/batect/config/DeviceMountSpec.kt

```diff
@@ -0,0 +1,227 @@
+/*
+   Copyright 2017-2019 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package batect.config
+
+import batect.config.io.deserializers.PathDeserializer
+import batect.os.PathResolutionResult
+import batect.os.PathType
+import batect.testutils.createForEachTest
+import batect.testutils.on
+import batect.testutils.osIndependentPath
+import batect.testutils.runForEachTest
+import batect.testutils.withColumn
+import batect.testutils.withLineNumber
+import batect.testutils.withMessage
+import com.charleskorn.kaml.Yaml
+import com.natpryce.hamkrest.absent
+import com.natpryce.hamkrest.and
+import com.natpryce.hamkrest.assertion.assertThat
+import com.natpryce.hamkrest.equalTo
+import com.natpryce.hamkrest.throws
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doAnswer
+import com.nhaarman.mockitokotlin2.mock
+import kotlinx.serialization.Decoder
+import kotlinx.serialization.modules.serializersModuleOf
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object DeviceMountSpec : Spek({
+    describe("a device mount") {
+        describe("deserializing from YAML") {
+            val pathDeserializer by createForEachTest {
+                mock<PathDeserializer> {
+                    on { deserialize(any()) } doAnswer { invocation ->
+                        val input = invocation.arguments[0] as Decoder
+
+                        when (val originalPath = input.decodeString()) {
+                            "/invalid" -> PathResolutionResult.InvalidPath(originalPath)
+                            else -> PathResolutionResult.Resolved(originalPath, osIndependentPath(originalPath), PathType.Other)
+                        }
+                    }
+                }
+            }
```

Comment:
Is this needed? I can't see where it's used

Reply:
Good catch. I copied some stuff from the `VolumeMountSpec`, may have not cleared enough out. I'll remove it now.